tests/Resources/ti.ui.ios.collisionbehavior.test.js

```diff
@@ -0,0 +1,209 @@
+		});
+
+		// TODO: add some sort of test for Padding object type?
+
+		it('defaults to all zeroes',  () => {
+			const collision = Titanium.UI.iOS.createCollisionBehavior();
+			should(collision.referenceInsets.top).eql(0);
+			should(collision.referenceInsets.bottom).eql(0);
+			should(collision.referenceInsets.left).eql(0);
+			should(collision.referenceInsets.right).eql(0);
+		});
+
+		// TODO: test setting some values and they persist
+	});
+
+	describe('.treatReferenceAsBoundary', () => {
+		it('is a Boolean',  () => {
+			const collision = Titanium.UI.iOS.createCollisionBehavior();
+			should(collision).have.a.property('treatReferenceAsBoundary').which.is.a.Boolean();
+		});
+
+		it('defaults to true',  () => {
+			const collision = Titanium.UI.iOS.createCollisionBehavior();
+			should(collision.treatReferenceAsBoundar).be.true();
+		});
+	});
+
+	describe('#addBoundary', () => {
+		it('is a Function',  () => {
+			const collision = Titanium.UI.iOS.createCollisionBehavior();
+			should(collision).have.a.property('addBoundary').which.is.a.Function();
+		});
+	});
+
+	describe('#addItem', () => {
+		it('is a Function',  () => {
+			const collision = Titanium.UI.iOS.createCollisionBehavior();
+			should(collision).have.a.property('addItem').which.is.a.Function();
+		});
+	});
+
+	describe('#removeAllBoundaries', () => {
+		it('is a Function',  () => {
+			const collision = Titanium.UI.iOS.createCollisionBehavior();
+			should(collision).have.a.property('removeAllBoundaries').which.is.a.Function();
+		});
+	});
+
+	describe('#removeBoundary', () => {
+		it('is a Function',  () => {
+			const collision = Titanium.UI.iOS.createCollisionBehavior();
+			should(collision).have.a.property('removeBoundary').which.is.a.Function();
+		});
+	});
+
+	describe('#removeItem', () => {
+		it('is a Function',  () => {
+			const collision = Titanium.UI.iOS.createCollisionBehavior();
+			should(collision).have.a.property('removeItem').which.is.a.Function();
+		});
+	});
+
+	describe('example', function () {
+		let win;
+
+		this.slow(2000);
+		this.timeout(10000);
+
+		afterEach(done => { // fires after every test in sub-suites too...
+			if (win && !win.closed) {
+				win.addEventListener('close', function listener () {
+					win.removeEventListener('close', listener);
+					win = null;
+					done();
+				});
+				win.close();
+			} else {
+				win = null;
+				done();
+			}
+		});
+
+		it('works', finish => {
+			win = Ti.UI.createWindow({
+				backgroundColor: 'white',
+				fullscreen: true
+			});
+
+			// Create an Animator object using the window as the coordinate system
+			const animator = Ti.UI.iOS.createAnimator({ referenceView: win });
+
+			// Create a default collision behavior, using the window edges as boundaries
+			const collision = Ti.UI.iOS.createCollisionBehavior();
+
+			// After 3 boundary collitions and 5 item collisions end the test
+			let boundaryCount = 3;
+			let itemCount = 5;
+			collision.addEventListener('itemcollision', e => {
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.ui.ios.collisionbehavior.test.js;
  Line: 156;
-->

- :warning: tests/Resources/ti.ui.ios.collisionbehavior.test.js line 156 – 'e' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)